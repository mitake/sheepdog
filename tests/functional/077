#!/bin/bash

# Test vdi repair functionality

. ./common

for i in `seq 0 2`; do
    _start_sheep $i
done

_wait_for_sheep 3

_cluster_format

$DOG vdi create test 12M
_vdi_list

echo "original data" | $DOG vdi write test

$DOG cluster shutdown

# single object lost

rm $STORE/0/obj/007c2b2500000000

for i in `seq 0 2`; do
    _start_sheep $i
done

_wait_for_sheep 3

$DOG vdi check test

$DOG vdi read test 0 14

$DOG cluster shutdown

# single broken object

_random | dd of=$STORE/1/obj/007c2b2500000000 bs=4096 count=1024 &> /dev/null

for i in `seq 0 2`; do
    _start_sheep $i
done

_wait_for_sheep 3

$DOG vdi check test

$DOG vdi read test 0 14

$DOG cluster shutdown

# two object lost

rm $STORE/0/obj/007c2b2500000000
rm $STORE/1/obj/007c2b2500000000

for i in `seq 0 2`; do
    _start_sheep $i
done

_wait_for_sheep 3

$DOG vdi check test

$DOG vdi read test 0 14

$DOG cluster shutdown

# single object lost, single broken object. no majority

rm $STORE/0/obj/007c2b2500000000
_random | dd of=$STORE/1/obj/007c2b2500000000 bs=4096 count=1024 &> /dev/null

for i in `seq 0 2`; do
    _start_sheep $i
done

_wait_for_sheep 3

$DOG vdi check test

$DOG cluster shutdown
