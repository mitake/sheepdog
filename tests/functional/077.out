QA output created by 077
using backend plain store
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag
  test         0   12 MB  0.0 MB  0.0 MB DATE   7c2b25     3              
fixed missing 7c2b2500000000
finish check&repair test
original data
fixed replica 7c2b2500000000
finish check&repair test
original data
fixed missing 7c2b2500000000
fixed missing 7c2b2500000000
finish check&repair test
original data
no majority of 7c2b2500000000
finish check&repair test
